reindexer/reindex_worker/src/test/scala/uk/ac/wellcome/platform/reindex_worker/services/ReindexWorkerServiceTest.scala

```diff
@@ -0,0 +1,216 @@
+import uk.ac.wellcome.sns.SNSWriter
+import uk.ac.wellcome.sqs.SQSReader
+import uk.ac.wellcome.storage.HybridRecord
+import uk.ac.wellcome.test.utils.{SNSLocal, SQSLocal}
+import uk.ac.wellcome.utils.JsonUtil._
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.concurrent.duration._
+
+class ReindexWorkerServiceTest
+    extends FunSpec
+    with Matchers
+    with DynamoDBLocal[HybridRecord]
+    with MockitoSugar
+    with SNSLocal
+    with SQSLocal
+    with ScalaFutures {
+
+  val actorSystem = ActorSystem()
+
+  val metricsSender: MetricsSender =
+    new MetricsSender(namespace = "reindexer-tests",
+                      mock[AmazonCloudWatch],
+                      actorSystem)
+
+  override lazy val tableName = "reindex-worker-service-test"
+
+  override lazy val evidence: DynamoFormat[HybridRecord] =
+    DynamoFormat[HybridRecord]
+
+  private val enrichedDynamoFormat: DynamoFormat[HybridRecord] = Sourced
+    .toSourcedDynamoFormatWrapper[HybridRecord]
+    .enrichedDynamoFormat
+
+  val queueUrl = createQueueAndReturnUrl("reindex-worker-service-test-q")
+  val topicArn = createTopicAndReturnArn("reindex-worker-service-test-topic")
+
+  it("returns a successful Future if the reindex completes correctly") {
+    val reindexJob = ReindexJob(
+      shardId = "sierra/123",
+      desiredVersion = 6
+    )
+
+    val hybridRecord = HybridRecord(
+      version = 1,
+      sourceId = "sierra",
+      sourceName = "111",
+      s3key = "s3://reindexWST/example.json",
+      reindexShard = reindexJob.shardId,
+      reindexVersion = reindexJob.desiredVersion - 1
+    )
+
+    Scanamo.put(dynamoDbClient)(tableName)(hybridRecord)(enrichedDynamoFormat)
+
+    val expectedRecords = List(
+      hybridRecord.copy(version = hybridRecord.version + 1,
+                        reindexVersion = reindexJob.desiredVersion)
+    )
+
+    val invalidSqsMessage = SQSMessage(
+      subject = None,
+      body = toJson(reindexJob).get,
+      topic = "topic",
+      messageType = "message",
+      timestamp = "now"
+    )
+
+    val service = reindexWorkerService()
+
+    val future = service.processMessage(message = invalidSqsMessage)
+
+    whenReady(future) { _ =>
+      val actualRecords: List[HybridRecord] =
+        Scanamo.scan[HybridRecord](dynamoDbClient)(tableName).map(_.right.get)
+
+      actualRecords shouldBe expectedRecords
+    }
+  }
+
+  it(
+    "returns a failed Future if it cannot parse the SQS message as a ReindexJob") {
+    val sqsMessage = SQSMessage(
+      subject = None,
+      body = "<xml>What is JSON.</xl?>",
+      topic = "topic",
+      messageType = "message",
+      timestamp = "now"
+    )
+
+    val service = reindexWorkerService()
+
+    val future = service.processMessage(message = sqsMessage)
+
+    whenReady(future.failed) { result =>
+      result shouldBe a[GracefulFailureException]
+      result.getMessage should include(
```

Comment:
I don't think we need to assert on the message

Reply:
I think it's worth asserting on the message here – we want to check we're failing on this particular error, and not somewhere else in the control flow.

Reply:
Oh yes, right I put another comment later. I think instead of checking the type and the messae of the exception, you can assert on what the exception is exactly. It is simpler and more clear